### PR TITLE
[BugFix] Iceberg supports array type

### PR DESCRIPTION
**Pull request by mxdzs0612**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by mxdzs0612** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 5 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|0|5|00.00%|[255, 256, 257, 259, 277]|

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java [hunk 1]
```diff
@@ -266,4 +272,8 @@ static Type convertColumnType(org.apache.iceberg.types.Type icebergType) {
     public static Database convertToSRDatabase(String dbName) {
         return new Database(CONNECTOR_DATABASE_ID_ID_GENERATOR.getNextId().asInt(), dbName);
     }
+
+    private static ArrayType convertToArrayType(org.apache.iceberg.types.Type icebergType) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java:276]: please add some unit tests

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 5 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|4|5|80.00%|[259]|

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java (hunk 1 above)
Comment (mxdzs0612) [reply fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java:276]: done.

File: fe/fe-core/src/test/java/com/starrocks/external/iceberg/IcebergUtilTest.java [hunk 2]
```diff
@@ -60,6 +61,14 @@ public void testString() {
         Assert.assertEquals(resType, stringType);
     }
 
+    @Test
+    public void testArray() {
```
Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/external/iceberg/IcebergUtilTest.java:65]: nit: add some corner cases

**Review by stephen-shelby — DISMISSED** [review]
LGTM

**Review by DorianZheng — DISMISSED** [review]
add ut for multiple list. eg, `list<list<........................list<int>>>>>>>>>`

rest LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 5 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|4|5|80.00%|[259]|

**Review by DorianZheng — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 5 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|4|5|80.00%|[259]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 5 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|4|5|80.00%|[259]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 5 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|4|5|80.00%|[259]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 5 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|4|5|80.00%|[259]|
